matrix_client/_api_asyncio.py

```diff
@@ -0,0 +1,78 @@
+import json
+from asyncio import sleep
+
+from matrix_client.api import MatrixHttpApi
+from matrix_client.errors import MatrixError, MatrixRequestError
+
+
+class AsyncHTTPAPI(MatrixHttpApi):
+    """
+    Contains all raw matrix HTTP client-server API calls using asyncio and aiohttp.
+
+    Usage:
+        async def main():
+            async with aiohttp.ClientSession() as session:
+                mapi = AsyncHTTPAPI("http://matrix.org", session)
+                resp = await mapi.get_room_id("#matrix:matrix.org")
+                print(resp)
+
+
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(main())
+    """
+
+    def __init__(self, base_url, client_session, token=None):
```

Comment:
I think being able to pass in a client session is definitely a nice thing, but
in the interest of simplicity, is there any way we could have a default
`client_session` if none is provided? This would require changing the sdk to
have an optional dependency on aiohttp; are optional dependencies a thing that
exists in python-land?

Reply:
The thing is the recommended way to create the session is with a context manager. I am not the sure of the best way to handle closing the session if it is created in the constructor of the class.

Reply:
Could you link me to some docs on the "sessions" you're talking about? Having a hard time understanding what cleanup is necessary.

Reply:
The first example on the first page for using aiohttp client uses `ClientSession` outside of a context manager, so I think we're okay on that front. Bigger question to me is if we want to tie ourselves to aiohttp as the default or if [asks](https://asks.readthedocs.io) is the better option with the way the ecosystem seems to be leaning away from asyncio and towards curio/trio.

Reply:
Actually I've flip-flopped on this again in the past 10 minutes. We should write the api class as generically as possible so that wrappers at the same abstraction level as `MatrixClient` can use whatever their preferred coroutine library is with this easily.